hudi-client/hudi-client-common/src/main/java/org/apache/hudi/index/HoodieIndexUtils.java

```diff
@@ -175,4 +186,68 @@ public static boolean checkIfValidCommit(HoodieTimeline commitTimeline, String c
     // 2) is less than the first commit ts in the timeline
     return !commitTimeline.empty() && commitTimeline.containsOrBeforeTimelineStarts(commitTs);
   }
+
+  public static <R> HoodieData<HoodieRecord<R>> getTaggedRecordsFromPartitionLocations(
```

Comment:
I feel we can do this in HoodieBloomIndexCheckFunction and HoodieKeyLookupHandle only and do it in one shot. as of cur patch, we fetch the locations based on just the base files and get the locations once and then call into getTaggedRecordsFromPartitionLocations which will again make getLatestFileSlice() and call snapshot read of base and log files to finalize record locations.